R/derived-variable-functions.R

```diff
@@ -1,36 +1,86 @@
-# Custom ifelse for evaluating NA
+#' Custom ifelse function that evaluates missing (NA) values
+#' 
+#' If the logical argument compares to a value that is NA, it will set `x` to `FALSE`  
+#' 
+#' @param x A logical argument
+#' 
+#' @param a value if `x` is `TRUE`
+#' 
+#' @param b value if `x` is `FALSE`
+#' 
+#' @return None
+#' 
+#' @examples ifelse2((age<18),"child", "adult")
+#' 
+#' @export
 ifelse2 <- function(x, a, b) {
   falseifNA <- function(x) {
     ifelse(is.na(x), FALSE, x)
   }
   ifelse(falseifNA(x), a, b)
 }
 
-#BMI derived variable
-# HWTGHTM: height (in meters)
-# HWTGWTK: weight (in kilograms)
+#' Body Mass Index (BMI) derived variable
+#' 
+#' This function creates a derived BMI variable independent from the BMI variable provided by the CCHS. 
+#' 
+#' This variable uses the CCHS variables for height and weight.
+#' 
+#' @param HWTGHTM CCHS variable for height (in meters)
+#' 
+#' @param HWTGWTK CCHS variable for weight (in kilograms)
+#' 
+#' @return a value for BMI
+#' 
+#' @export
 BMI_derived <- 
   function(HWTGHTM, 
            HWTGWTK) {
     ifelse2((!is.na(HWTGHTM)) & (!is.na(HWTGWTK)), 
             (HWTGWTK/(HWTGHTM*HWTGHTM)), NA)
   }
 
-#Smoking variables
```

Comment:
Suggest splitting the variables up into different. .R functions. One for BMI. One for smoking variables, etc.